tests/network/primary_pod_network.go

```diff
@@ -83,17 +84,55 @@ var _ = SIGDescribe("[Serial]Primary Pod Network", func() {
 		})
 
 		Context("VMI connected to the pod network using masquerade binding", func() {
-			var vmi *v1.VirtualMachineInstance
+			When("Guest Agent exists", func() {
+				var vmi *v1.VirtualMachineInstance
+
+				BeforeEach(func() {
+					networkData, err := libnet.CreateDefaultCloudInitNetworkData()
+					Expect(err).NotTo(HaveOccurred())
+
+					tmpVmi := libvmi.NewFedora(
```

Comment:
Why aren't you using `tests.NewRandomFedoraVMIWitGuestAgent` it does the same.

Reply:
`tests.NewRandomFedoraVMIWitGuestAgent` is using `libvmi`. It has exactly the same code you have here.
Do you plan to remove `tests.NewRandomFedoraVMIWitGuestAgent`? If yes, I would move it to libvmi. I don't think that all (16) usages of it should be substituted with the exact same 5 code lines.

Reply:
My plan is not to use it because that function does not allow me to configure network stuff in parallel (it uses internally a default cloud-init network config). So I would just not use it in any new code, especially in the network sig.
This is the reason I think it does not fit any global helper (e.g. libvmi helper).

Removing it is something I planned indeed, just to make sure no one will use it by default. I wanted to just move it into the tests as a helper there (per file).

Reply:
But here in this case you use the default network config. We see that there is a need for such config. There are 16 places which are using `tests.NewRandomFedoraVMIWitGuestAgent`. Why not having a default method for creating a vmi with guest agent.
Can you please explain how duplicating this method to 3 or more files (per file using the method) is better than writing it once?

Reply:
As I see it, if you have a helper that fits one specific usage but limits other usages, it does not belong to a global usage because it will later on collide.
I tried to give in this specific case the exact case where this will happen: If we need to create a VMI with the Guest Agent, then also add IPv6 configuration, we will not be able to use this helper and it will surely collide with anything we may need.

The same reasoning to re-use the same config by creating functions for each and every combination one can think of caused details to be placed in multiple bastraction layers causing the caller to just re-apply some just to make sure no one adds them down the stack. This is the case which the libvmi package came to solve, trying to flatten the creating of the vmi spec.

We may end up with `NewRandomFedoraVMIWithGuesAgentWithDefaultNetwork` and each time we need something else we will add to the name more and more, getting back to the current problem of too many functions representing all kind of combinations. I am trying to avoid it.

I am also claiming this is usually more expressive to detail from what building blocks the vmi is created from.
I am also trying to say that as long that the abstraction is not limiting (e.g. the NewFedora factory), we can use it.

Reply:
The NewFedora factory is limiting as well. If you want bridge binding for example, you cannot use it. I don't understand the big difference.
As a compromise, please move the "vm with guest agent" creation to a method in the primary_pod_network.go file.